packages/cli/tests/e2e/bot/FuncHostedNotificationHappyPathCommon.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @author Siglud
+ */
+
+import * as fs from "fs-extra";
+import * as path from "path";
+import { BotValidator } from "../../commonlib";
+import {
+  cleanUp,
+  execAsync,
+  execAsyncWithRetry,
+  getSubscriptionId,
+  getTestFolder,
+  getUniqueAppName,
+  readContextMultiEnv,
+} from "../commonUtils";
+import { environmentManager } from "@microsoft/teamsfx-core";
+
+import { Runtime } from "../../commonlib/constants";
+
+export async function happyPathTest(runtime: Runtime) {
+  const testFolder = getTestFolder();
+  const appName = getUniqueAppName();
+  const subscription = getSubscriptionId();
+  const projectPath = path.resolve(testFolder, appName);
+  const envName = environmentManager.getDefaultEnvName();
+
+  const env = Object.assign({}, process.env);
+  env["TEAMSFX_CONFIG_UNIFY"] = "true";
+  env["BOT_NOTIFICATION_ENABLED"] = "true";
+  env["TEAMSFX_TEMPLATE_PRERELEASE"] = "beta";
+  if (runtime === Runtime.Dotnet) {
+    env["TEAMSFX_CLI_DOTNET"] = "true";
+  }
+
+  const cmdBase = `teamsfx new --interactive false --app-name ${appName} --capabilities notification --bot-host-type-trigger http-functions`;
```

Comment:
Better to add timer-functions test case as well.

Reply:
I think the provision and deploy process is the same for the timer function and the HTTP functions